gprofiler/profilers/factory.py

```diff
@@ -0,0 +1,38 @@
+from typing import TYPE_CHECKING, Any, List, Tuple, Union
+
+from gprofiler.log import get_logger_adapter
+from gprofiler.profilers.profiler_base import NoopProfiler
+
+if TYPE_CHECKING:
+    from gprofiler.main import UserArgs
+    from gprofiler.profilers.profiler_base import ProcessProfilerBase
+
+from gprofiler.profilers.perf import SystemProfiler
+from gprofiler.profilers.registry import get_profilers_registry
+
+logger = get_logger_adapter(__name__)
+
+
+def get_profilers(
+    user_args: 'UserArgs', **profiler_init_kwargs: Any
+) -> Tuple[Union['SystemProfiler', 'NoopProfiler'], List['ProcessProfilerBase']]:
+    profilers_registry = get_profilers_registry()
+    process_profilers_instances: List['ProcessProfilerBase'] = []
+    system_profiler: Union['SystemProfiler', 'NoopProfiler'] = NoopProfiler()
+    for profiler_name, profiler_config in profilers_registry.items():
+        profiler_mode = user_args.get(f"{profiler_name.lower()}_mode")
+        if profiler_mode in ("none", "disabled") or not user_args.get(profiler_name.lower(), True):
+            continue
+        try:
+            profiler_instance = profiler_config.profiler_class(
+                **profiler_init_kwargs, **user_args, profiler_mode=profiler_mode
+            )
+        except Exception:
+            logger.exception(f"Couldn't create {profiler_name} profiler, continuing without this runtime profiler")
```

Comment:
@Jongy Please note that if the perf initialization fails, we will continue in a "non-perf" mode. Want me to change the behavior and crash instead?

Reply:
Yes. That was the behvaior (e.g, in 1.1.3 it crashes). We (or I :) kinda broke it in 1.1.7, now it fails with some `TimeoutError` and doesn't print the error from `perf`, which is pretty indicative (`perf_event_open(..., PERF_FLAG_FD_CLOEXEC) failed with unexpected error 1 (Operation not permitted)`).

Please fix your code to crash as it used to. And also fix gProfiler to print the proper error message.

Reply:
Will run without `--privileged`

Reply:
Tried?

Reply:
It didn't work as it failed on another area, so I manually added a `raise Exception()` to the `SystemProfiler.__init__` method. It stopped the agent from booting.

Reply:
good